### ISSUE TITLE
Support f64

### ISSUE THREAD

**Issue by alexheretic** [orhanbalci/rust-easing#7]
Hi thanks for writing up these easing functions in rust! I would like support for f64 precision, as all the function explicitly use f32. I think having the option of both precisions would be useful for different situations.

It could work something like
```rust
use easer::functions::*;
let single: f32 = Circ::ease_in(2f32, 0f32, 50f32, 4f32));
// add new f64 variants of each easing function
let double: f64 = Circ64::ease_in(2.0, 0.0, 50.0, 4.0)); 
```

However, It might actually be better to follow the rust languages defaulting to f64 instead, ie:
```rust
// rename current functions to *32
let single: f32 = Circ32::ease_in(2f32, 0f32, 50f32, 4f32));
// add new f64 variants of each easing function with the current names
let double: f64 = Circ::ease_in(2.0, 0.0, 50.0, 4.0)); 
```
I was thinking this could be achieved with a macro, to avoid duplication. 
Would you be up for one of these? I can raise a PR if you like.

**Comment by orhanbalci** [comment]
Hi @alexheretic 
I was thinking about a generic implementation using num
crate with num::float type.  If you are interested you may give a shot.

**Comment by alexheretic** [comment]
Cool I'll have a look at implementing something tomorrow.

**Comment by alexheretic** [comment]
Solved by #8
